### PR TITLE
Issue 12033 - Use version of habitica-markdown that includes mentions

### LINKED ISSUE #12033: Usernames with underscores on each side aren't displayed correctly
If a player has a Username that starts and ends with an underscore (e.g., `@_spider_`), when their Username is mentioned in a chat post, there are three formatting problems:

- the name is in italics (that's kind of expected because underscores indicate italics in markdown but markdown parsing within a Username is not desired) 
- the underscores are not visible (a side-effect of the previous point but this is undesirable because you're not seeing the player's proper Username)
- the text colour of Username is wrong (screenshot below)

The screenshot below was taken from a post that contained this raw text:
`@_spider_ and @blackgoblingo`
Both of those Usernames were in use at the time I made the post (the `_spider_` user has a different name now).
The formatting for the `blackgoblingo` user is correct.
The formatting for the `_spider_` user is wrong.

Both usernames do correctly point to the right profile screens, so that's something. :)

The expected behaviour is that you see `@_spider_` and that it's formatted in the same way as the `blackgoblingo` username.

image

_Thanks to @EliotThe (0d77107e-9128-4656-8c02-6e06331fa396) for reporting this!_

### PR DESCRIPTION
**Pull request by benkelaar**
Fixes #12033

Also fixes frontend parts of #11504 and #10924 

~Is in draft since it is dependent on https://github.com/HabitRPG/habitica-markdown/pull/87 to be merged first.~

~Edit: Now also dependent on #12069 -> Please merge that one before reviewing this one, this branch is dependent on that one and as such this diff view is polluted with the contents of that branch.~

### Changes
- Updated the dependency on habitica-markdown to 2.0.0 for both server and client.
- Removed the highlightUser method
- Updated the cards and tests to use markdown directly with passed in username and displayname

----
UUID: d71d2c57-a73d-4591-b64d-e688584a9092

### CODE REVIEW

File: website/client/package-lock.json [hunk 1]
```diff
@@ -11676,7 +11676,7 @@
       }
     },
     "habitica-markdown": {
-      "version": "1.4.0",
```
Comment (paglias) [inline website/client/package-lock.json:11679]: I think you'll have to manually install `habitica-markdown@2` with `npm i habitica-markdown@2` or the lockfile won't be updated properly

File: website/client/package-lock.json [hunk 2]
```diff
@@ -11676,7 +11676,7 @@
...
+      "version": "2.0.0",
       "resolved": "https://registry.npmjs.org/habitica-markdown/-/habitica-markdown-1.4.0.tgz",
```
Comment (paglias) [inline website/client/package-lock.json]: I think you'll have to manually install habitica-markdown@2 with npm i habitica-markdown@2 or the lockfile won't be updated properly

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/client/package-lock.json (hunk 2 above)
Comment (benkelaar) [reply website/client/package-lock.json]: Yeah I think it went wrong because the version wasn't available at the time. That situation has improved now! 🥳

**Comment by paglias** [comment]
Tested it a bit and looks like nothing broke 🎉 I guess all the markdown stuff is done for now, thanks for all the PRs @benkelaar !

**Comment by benkelaar** [comment]
Wow, genuinely feels like an accomplishment! Like I said in the other PR, thanks a lot for your support! The team is doing a great job helping out a new contributor like myself!

Now all we need to do is sit back and wait for the bug reports to come rolling in 😉. Or do one of the other issues... Or spoiler tags... Or ninja emoji... Or emoji suggestion in web... Or link detection in Android... Or markdown previews... Or WYSIWIG... Or... Always plenty to do for a blacksmith in habitica! 😁

**Pull request merged**